src/Middleware/ConcurrencyLimiter/src/QueuePolicies/QueuePolicy.cs

```diff
@@ -10,49 +10,49 @@ namespace Microsoft.AspNetCore.ConcurrencyLimiter
 {
     internal class QueuePolicy : IQueuePolicy, IDisposable
     {
-        private readonly int _maxConcurrentRequests;
-        private readonly int _requestQueueLimit;
+        private readonly int _maxTotalRequest;
         private readonly SemaphoreSlim _serverSemaphore;
 
         private object _totalRequestsLock = new object();
+
+        private readonly static ValueTask<bool> _falseValueTask = new ValueTask<bool>(false);
+
         public int TotalRequests { get; private set; }
 
         public QueuePolicy(IOptions<QueuePolicyOptions> options)
         {
-            _maxConcurrentRequests = options.Value.MaxConcurrentRequests;
-            if (_maxConcurrentRequests <= 0)
+            if (options.Value.MaxConcurrentRequests <= 0)
             {
-                throw new ArgumentException(nameof(_maxConcurrentRequests), "MaxConcurrentRequests must be a positive integer.");
+                throw new ArgumentException(nameof(options.Value.MaxConcurrentRequests), "MaxConcurrentRequests must be a positive integer.");
             }
 
-            _requestQueueLimit = options.Value.RequestQueueLimit;
-            if (_requestQueueLimit < 0)
+            if (options.Value.RequestQueueLimit < 0)
             {
-                throw new ArgumentException(nameof(_requestQueueLimit), "The RequestQueueLimit cannot be a negative number.");
+                throw new ArgumentException(nameof(options.Value.RequestQueueLimit), "The RequestQueueLimit cannot be a negative number.");
             }
 
-            _serverSemaphore = new SemaphoreSlim(_maxConcurrentRequests);
+            _serverSemaphore = new SemaphoreSlim(options.Value.MaxConcurrentRequests);
+
+            _maxTotalRequest = options.Value.MaxConcurrentRequests + options.Value.RequestQueueLimit;
         }
 
-        public async ValueTask<bool> TryEnterAsync()
+        public ValueTask<bool> TryEnterAsync()
```

Comment:
I also think removing the await here isn't a necessary optimization. The more common path here will be to await the semaphore; returning _falseValueTask will happen effectively never. 

Therefore, adding an extra function call to await the semaphore most likely slows down performance.

Reply:
You could keep this pattern, but instead, check if `_serverSemaphore.WaitAsync()` is completed synchronously or not"
```c#
var task = _serverSemaphore.WaitAsync();
if (task.IsCompletedSuccessfully)
{
    return new ValueTask<bool>(true);
}
```